independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/logging/QuarkusDelayedHandler.java

```diff
@@ -238,6 +238,7 @@ public void removeHandler(final Handler handler) throws SecurityException {
         for (Runnable i : logCloseTasks) {
             i.run();
         }
+        logCloseTasks.clear();
```

Comment:
Only half related, but I think it makes sense to dispose the tasks that were executed.

Reply:
Indeed